### PR TITLE
[WIP] fix(#5658): handle edge cases for tag filtering detection

### LINKED ISSUE #5658: Startup issue with KIC wipes database
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

In a certain startup scenario, KIC nukes all pre-existing config in the Kong Admin API that's not managed by it due to how tag filtering gets configured.

Possibly related to https://github.com/Kong/kubernetes-ingress-controller/issues/2274

### Expected Behavior

Tag filtering should be on by default to avoid removing pre-existing config in case of failure.

### Steps To Reproduce

```markdown
1. In a Kubernetes cluster, with:

- Kong Ingress Controller (3.1) deployed via helm (2.34.0) without any FeatureGates enabled
- Kong Gateway (3.6) deployed via helm (2.34.0)
- Those two are deployed in separate helm releases and not via the kong ingress helm chart
- Gateway API deployed (https://github.com/kubernetes-sigs/gateway-api/releases/download/v1.0.0/experimental-install.yaml)


2. Create Kong Consumer / other Kong resources that are do not have the `managed-by-ingress-controller` tag via the Admin API

3. Scale down the gateway API to 0 replicas. 

4. Rerollout the Ingress Controller and observe the logs of the elected leader until the following log appears: `tag filtering disabled because Kong Admin API does not support tags`

5. Rescale up the Kong Gateway. 

6. Observe as it wipes non Ingress Controller managed configs and removes the `managed-by-ingress-controller` tag from all currently managed objects
```

### Kong Ingress Controller version

```shell
Kong Ingress Controller (3.1) 
Kong Gateway (3.6)
```

### Kubernetes version

```shell
Independent of k8s version
```

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by ichbinfrog**
**What this PR does / why we need it**:

This PR makes detecting if tags filtering is enabled more robust. If the admin API is unreachable, the controller will fail fast rather then disabling tags (which can have undesired consequences, e.g removing objects not managed by the controller).

**Which issue this PR fixes**:

fixes #5658

**Special notes for your reviewer**:

I'm not sure how to add e2e tests for this change since it requires simulating a connection failure during initialization both when the admin api supports tags and does not. 

**PR Readiness Checklist**:

Complete these before marking the PR as `ready to review`:

- [x] the `CHANGELOG.md` release notes have been updated to reflect any significant (and particularly user-facing) changes introduced by this PR

### CODE REVIEW

**Comment by rainest** [comment]
I think we can probably do this simpler by just removing the tag support check altogether.

The original introduction of that check is [quite old](https://github.com/Kong/kubernetes-ingress-controller/pull/1415), from an era where there were (maybe? even that long ago I don't think we really supported anything prior to gateway 1.0) gateway versions without tag support in the wild.

3.x releases of KIC cut off support for all gateway releases prior to 3.4, so everything we currently support in the controller supports tags. I'll double check to see if there's any other reason to keep this, but I suspect we can just discard it and always enable tag filtering.

**Comment by ichbinfrog** [comment]
Thanks for looking into it @rainest 🙏 , whilst you double check if the tag check can be completely removed, I'll deploy this test build in our cluster to see if the issue with consumers getting deleted.

**Comment by ichbinfrog** [comment]
Small update, the patch deployment made the issue disappear from the Kong Ingress Controller in our Kubernetes cluster. I've also taken the liberty to remove code related to checking tags as you suggested

File: CHANGELOG.md [hunk 1]
```diff
@@ -84,6 +84,13 @@ Adding a new version? You'll need three changes:
  - [0.0.5](#005)
  - [0.0.4 and prior](#004-and-prior)
 
+## Unreleased
+
+### Fixed
+
+- When initializing the controller, if the admin API is unreachable, the controller
+  fails fast rather than disabling tags. [#5658](https://github.com/Kong/kubernetes-ingress-controller/issues/5658)
```
Comment (rainest) [inline CHANGELOG.md:92]:
```suggestion
- Remove unnecessary tag support check that could incorrectly delete configuration if the check did not execute properly. [#5658](https://github.com/Kong/kubernetes-ingress-controller/issues/5658)
```

**Review by rainest — CHANGES_REQUESTED** [review]
Should be good to go, just needs an updated changelog entry that reflects the final change.

**Review by rainest — APPROVED** [review]

**Pull request merged**